api/catalog/api/controllers/search_controller.py

```diff
@@ -132,6 +158,27 @@ def _post_process_results(
         validate_images(query_hash, start, results, to_validate)
 
         if len(results) < page_size:
+            """
+            The variables in this function get updated in an interesting way.
+            Here is an example of that for a typical query. Note that ``end``
+            increases but start stays the same. This has the effect of slowly
+            increasing the size of the query we send to Elasticsearch with the
+            goal of backfilling the results until we have enough valid (live)
+            results to fulfill the requested page size.
+
+            ```
+            page_size: 20
+            page: 1
+
+            start: 0
+            end: 40 (DEAD_LINK_RATIO applied)
+
+            end gets updated to end + end/2 = 60
+
+            end = 90
+            end = 90 + 45
```

Comment:
Slight rewording here, the 60+30 operation is missing and this makes it a bit clearer IMO.
```suggestion
            end gets updated to end + end/2 = 60
            end = 60 + 30 = 90
            end = 90 + 45 = 135
```